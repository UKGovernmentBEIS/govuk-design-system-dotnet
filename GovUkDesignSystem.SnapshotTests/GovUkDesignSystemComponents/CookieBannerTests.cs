using System.Collections.Generic;
using System.Threading.Tasks;
using GovUkDesignSystem.GovUkDesignSystemComponents;
using GovUkDesignSystem.GovUkDesignSystemComponents.SubComponents;
using GovUkDesignSystem.SnapshotTests.Helpers;
using Xunit;

namespace GovUkDesignSystem.SnapshotTests.GovUkDesignSystemComponents
{
    public class CookieBannerTests : SnapshotTestBase
    {
        private CookieBannerViewModel DefaultCookieBannerViewModel()
        {
            return new CookieBannerViewModel
            {
                AriaLabel = "Cookies on unnamed service",
                Classes = "banner-classes",
                Messages = new List<CookieBannerMessageViewModel>
                {
                    new CookieBannerMessageViewModel
                    {
                        Actions = new List<CookieBannerMessageActionViewModel>
                        {
                            new CookieBannerMessageActionViewModel
                            {
                                Classes = "action-classes",
                                Href = "#",
                                Text = "Button text",
                                Type = "button"
                            }
                        },
                        Classes = "message-classes",
                        HeadingText = "Heading text",
                        Role = "Banner",
                        Text = "Message text"
                    }
                }
            };
        }

        [Fact]
        public async Task Render_AllValues()
        {
            // Act & Assert
            await VerifyPartial("CookieBanner", DefaultCookieBannerViewModel());
        }
    }
}