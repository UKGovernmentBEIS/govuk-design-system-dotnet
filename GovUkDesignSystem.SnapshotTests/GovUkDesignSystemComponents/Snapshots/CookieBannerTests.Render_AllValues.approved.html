
<div class="govuk-cookie-banner banner-classes" data-nosnippet role="region" aria-label="Cookies on unnamed service"
     
     >
    

<div class="govuk-cookie-banner__message govuk-width-container message-classes"
     role=&quot;Banner&quot;
     
     >
     
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
                <h2 class="govuk-cookie-banner__heading govuk-heading-m">

Heading text                </h2>
            
            <div class="govuk-cookie-banner__content">

Message text            </div>
            
                <div class="govuk-button-group">



        <a class="govuk-button action-classes"
           data-module="govuk-button"
           
                   href="#"
                          role="button"
           draggable="false"
           >

Button text        </a>
                </div>
        </div>
    </div>
</div></div>